ee/hogai/graph/root/nodes.py

```diff
@@ -629,6 +643,36 @@ def _render_session_summarization_context(self, config: RunnableConfig) -> str:
 
 
 class RootNodeTools(AssistantNode):
+    async def get_reasoning_message(
+        self, input: BaseState, default_message: Optional[str] = None
+    ) -> ReasoningMessage | None:
+        if not isinstance(input, BaseStateWithMessages):
+            return None
+        assert isinstance(input.messages[-1], AssistantMessage)
+        tool_calls = input.messages[-1].tool_calls or []
+        assert len(tool_calls) <= 1
+        if len(tool_calls) == 0:
+            return None
+        tool_call = tool_calls[0]
+        content = None
+        if tool_call.name == "create_and_query_insight":
+            content = "Coming up with an insight"
+        elif tool_call.name == "search_documentation":
+            content = "Checking PostHog docs"
+        elif tool_call.name == "retrieve_billing_information":
+            content = "Checking your billing data"
+        else:
+            # This tool should be in CONTEXTUAL_TOOL_NAME_TO_TOOL, but it might not be in the rare case
+            # when the tool has been removed from the backend since the user's frontend was loaded
+            ToolClass = CONTEXTUAL_TOOL_NAME_TO_TOOL.get(tool_call.name)  # type: ignore
```

Comment:
Seems hacky, but yeah, can't propose any better solution 😅 

A question though - I assume `CONTEXTUAL_TOOL_NAME_TO_TOOL` expects `AssistantContextualTool` enum as a key, and we are sure that `tool_call.name` will be one? Or we don't, that why `ignore`?

Reply:
Also code from before this PR, but anyways the `.get()` returns None if the tool_call.name is not in the `AssistantContextualTool` enum